src/library-authoring/LibraryAuthoringPage.test.tsx

```diff
@@ -0,0 +1,236 @@
+import React from 'react';
+import MockAdapter from 'axios-mock-adapter';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { AppProvider } from '@edx/frontend-platform/react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import fetchMock from 'fetch-mock-jest';
+
+import initializeStore from '../store';
+import { getContentSearchConfigUrl } from '../search-modal/data/api';
+import mockResult from '../search-modal/__mocks__/search-result.json';
+import mockEmptyResult from '../search-modal/__mocks__/empty-search-result.json';
+import LibraryAuthoringPage from './LibraryAuthoringPage';
+import { getContentLibraryApiUrl } from './data/api';
+
+let store;
+const mockUseParams = jest.fn();
+let axiosMock;
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
+  useParams: () => mockUseParams(),
+}));
+
+const searchEndpoint = 'http://mock.meilisearch.local/multi-search';
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
+
+const returnEmptyResult = (_url, req) => {
+  const requestData = JSON.parse(req.body?.toString() ?? '');
+  const query = requestData?.queries[0]?.q ?? '';
+  // We have to replace the query (search keywords) in the mock results with the actual query,
+  // because otherwise we may have an inconsistent state that causes more queries and unexpected results.
+  mockEmptyResult.results[0].query = query;
+  // And fake the required '_formatted' fields; it contains the highlighting <mark>...</mark> around matched words
+  // eslint-disable-next-line no-underscore-dangle, no-param-reassign
+  mockEmptyResult.results[0]?.hits.forEach((hit) => { hit._formatted = { ...hit }; });
+  return mockEmptyResult;
+};
+
+const libraryData = {
```

Comment:
```suggestion
const libraryData: ContentLibrary = {
```

and change the import above to `import { getContentLibraryApiUrl, type ContentLibrary } from './data/api';`

Because you have some type errors here, like `allowPublic_learning` should be `allowPublicLearning`

Reply:
Fixed here: 4d9da7f4e02935e0316bcce33ca84b75dce2bcf5